corgi/api/views.py

```diff
@@ -380,12 +381,49 @@ class ComponentViewSet(ReadOnlyModelViewSet):  # TODO: TagViewMixin disabled unt
     filterset_class = ComponentFilter
     lookup_url_kwarg = "uuid"
 
+    def get_queryset(self):
+        mode = self.request.query_params.get("mode")
+        if mode == "latest":
+            # 'latest' mode only relevent in terms of a specific offering/product
+            # TODO - subject of a near-future PR will handle instances of when we have
+            #      multiple root components with the same component name yet different versions.
+            ofuri = self.request.query_params.get("ofuri")
+            if ofuri:
+                component_types = Q(type=Component.Type.SRPM) | Q(
+                    type=Component.Type.CONTAINER_IMAGE
+                )
+                results = (
+                    Component.objects.filter(
+                        component_types,
+                        product_streams__overlap=[ofuri],
+                        software_build__completion_time__isnull=False,
+                    )
+                    .annotate(
+                        latest=Subquery(
+                            Component.objects.filter(
+                                component_types,
+                                name=OuterRef("name"),
+                                product_streams__overlap=[ofuri],
+                                software_build__completion_time__isnull=False,
```

Comment:
Are all the filters besides `name=OuterRef("name")` actually necessary? I think the parent query already pairs it down to the components that we need.

Reply:
when we do not include these the ordered list includes outside of original parent query ... which then results in a wrong answer ... let me retest again to verify this is the case (possible I was confused)

Reply:
removed